cmd/fleet/serve.go

```diff
@@ -263,6 +263,35 @@ the way that the Fleet server works.
 				}
 			}
 
+			if config.Packaging.GlobalEnrollSecret != "" {
+				secrets, err := ds.GetEnrollSecrets(cmd.Context(), nil)
+				if err != nil {
+					initFatal(err, "loading enroll secrets")
+				}
+
+				var gotSecret bool
+				for _, secret := range secrets {
+					if secret.TeamID == nil {
+						gotSecret = true
+						break
+					}
+				}
+
+				if gotSecret {
```

Comment:
Shouldn't this be `globalEnrollSecret != "" && globalEnrollSecret != config.Packaging.GlobalEnrollSecret` (using HEAD^ variable names)

Reply:
There's the case where a fleet restart with the same config value should not print the warning, right?